File: app/src/main/java/com/example/brauliomendez/cleanarquitecturedaggerandgithubapi/search/domain/usecase/GithubSearchRepositoriesUseCase.java
package com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.search.domain.usecase;


import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.search.domain.entity.SearchResponse;
import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.search.domain.repository.GithubSearchRepository;

import io.reactivex.Observable;

/**
 * Created by Leo on 24/01/17.
 */

public class GithubSearchRepositoriesUseCase {

  private GithubSearchRepository githubSearchRepository;

  public GithubSearchRepositoriesUseCase(
      GithubSearchRepository githubSearchRepository) {
    this.githubSearchRepository = githubSearchRepository;
  }

  public Observable<SearchResponse> searchGithubRepositories(String username){
    return githubSearchRepository.searchGithubRepositories(username);
  }
}

File: kotlinversion/src/main/java/com/example/kotlinversion/domain/usecase/GithubSearchRepositoriesUseCase.kt
package com.example.kotlinversion.domain.usecase

import com.example.kotlinversion.domain.entity.SearchResponse
import com.example.kotlinversion.domain.repository.GithubSearchRepository
import io.reactivex.Observable

/**
 * Created by BraulioMendez on 1/29/17.
 */
class GithubSearchRepositoriesUseCase(var githubSearchRepository: GithubSearchRepository) {

    fun githubSearchRepositories(username : String) : Observable<SearchResponse> {
        return githubSearchRepository.searchGithubRepositories(username)
    }
}